packages/react-combobox/src/utils/OptionCollection.types.ts

```diff
@@ -13,32 +8,25 @@ export type OptionValue = {
 
 export type OptionCollectionValue = {
   /** The total number of options in the collection. */
-  count: number;
+  count: () => number;
```

Comment:
Nit: count feels like a property at first glance to me.  Consider renaming to getCount (maybe a later PR?).